trackintel/preprocessing/triplegs.py

```diff
@@ -145,31 +149,40 @@ def generate_trips(spts, tpls, gap_threshold=15):
     spts_tpls_no_act = spts_tpls[~spts_tpls["activity"]]
     spts_tpls_only_act = spts_tpls[spts_tpls["activity"]]
 
+    # todo: maybe move this part down to where you need it
     # add gaps as activities, to simplify id assignment later.
     gaps = pd.DataFrame(spts_tpls.loc[gap, "user_id"])
     gaps["started_at"] = spts_tpls.loc[gap, "finished_at"] + gap_threshold / 2
     gaps[["type", "activity", "new_trip"]] = ["gap", True, True]
 
     # same for user changes
     user_change = pd.DataFrame(spts_tpls.loc[condition_new_user, "user_id"])
+    # todo: if the started_at will be used for sorting then the time difference should be as small as possible
     user_change["started_at"] = spts_tpls.loc[condition_new_user, "started_at"] - gap_threshold / 2
     user_change[["type", "activity", "new_trip"]] = ["user_change", True, True]
 
     trips_grouper = spts_tpls_no_act.groupby("trip_id")
+    # todo: for me using `np.array` raised an exception: "ValueError: Must produce aggregated value"
```

Comment:
That is strange. Maybe they updated that in a newer pandas version. https://stackoverflow.com/questions/16975318/pandas-aggregate-when-column-contains-numpy-arrays

Reply:
Good fix. :)